### PR TITLE
<FacesRatingBar /> - new component

### PR DESCRIPTION
**Pull request by sivanharel**
### 🔦 Summary

...

### ✅ Checklist

- [ ] 👨‍💻 API change is approved by the UI Infra Developers 
- [ ] 👨‍🎨 UX change is approved by the Design System UX 
- 📚 Change is documented
  - [ ] Story
  - [ ] API description
  - [ ] Cheatsheet
  - [ ] Other (explain)
- 🔬 Change is tested
  - [ ] Component
  - [ ] Visual test

### CODE REVIEW

**Comment by WixBuildServer** [comment]
### 🤖 Hi, I'm #Janet!

  A deploy preview of your PR was successfully released 🖼.

  You can find it under one of the following URLs:
  * https://www.wix.com/pages/wix-style-react.pr_5643/

File: perfer.config.js [hunk 1]
```diff
@@ -141,6 +141,7 @@ const javascriptFiles = [
   ['VerticalTabs.bundle.min.js', 38],
   ['VerticalTabsItem.bundle.min.js', 38],
   ['WarningIndicator.bundle.min.js', 40],
+  ['FacesRatingBar.bundle.min.js', 37],
```
Comment (EgoziE) [inline perfer.config.js:144]: sort

File: src/FacesRatingBar/test/FacesRatingBar.visual.js [hunk 2]
```diff
@@ -0,0 +1,88 @@
+import React from 'react';
+import { storiesOf } from '@storybook/react';
+import FacesRatingBar from '../FacesRatingBar';
+
+const commonProps = {
+  value: 2,
+};
+
+const tests = [
```
Comment (EgoziE) [inline src/FacesRatingBar/test/FacesRatingBar.visual.js:9]: Is something tested here?

File: src/FacesRatingBar/FacesRatingBar.js [hunk 3]
```diff
@@ -0,0 +1,275 @@
...
+  hoveredIndex,
+  showDescriptionValues,
+  descriptionValues,
+  onClick,
+  onMouseEnter,
+  onMouseLeave,
+  handleFocus,
+  handleBlur,
+}) => {
+  return Object.values(faceIndexes).map(faceIndex => {
```
Comment (EgoziE) [inline src/FacesRatingBar/FacesRatingBar.js:125]: Optional

```suggestion
  return Array(5).fill(0).map((_, index)=> {
```
and then use the index
or use a common for loop

File: src/FacesRatingBar/constants.js [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+  2: 'frowningTooltip',
+  3: 'neutralTooltip',
+  4: 'smilingTooltip',
+  5: 'griningTooltip',
+};
+
+export const facesRatingBarSizes = {
+  medium: 'medium',
+};
+
```
Comment (EgoziE) [inline src/FacesRatingBar/constants.js:21]: never used.

File: src/FacesRatingBar/constants.js [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+export const faceIconTypeMap = {
+  1: 'disapointed',
+  2: 'frowning',
+  3: 'neutral',
+  4: 'smiling',
+  5: 'grining',
+};
+
```
Comment (EgoziE) [inline src/FacesRatingBar/constants.js:37]: You can have one data object:

```suggestion
export const facesData = {
  1: { name: 'disapointed', dataHook: 'face-disapointed' },
  2: .....,
  3: .....,
  4: .....,
  5: .....,
};
```

File: src/FacesRatingBar/test/FacesRatingBar.spec.js [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+import { facesRatingBarPrivateDriverFactory } from './FacesRatingBar.private.uni.driver';
+
+describe(FacesRatingBar.displayName, () => {
+  const render = createRendererWithUniDriver(
+    facesRatingBarPrivateDriverFactory,
+  );
+
+  afterEach(() => {
+    cleanup();
+  });
```
Comment (EgoziE) [inline src/FacesRatingBar/test/FacesRatingBar.spec.js:14]:
```suggestion
  afterEach(cleanup);
```

File: src/FacesRatingBar/FacesRatingBar.js (hunk 3 above)
Comment (sivanharel) [reply src/FacesRatingBar/FacesRatingBar.js:103]: I changed it but I think it was more readable before these changes

File: src/FacesRatingBar/FacesRatingBar.js [hunk 7]
```diff
@@ -0,0 +1,327 @@
...
+        iconType: iconType,
+      })}
+    >
+      <IconTagName />
+    </div>
+  );
+};
+
+// faces:
+const FaceDisapointed = () => {
```
Comment (EgoziE) [inline src/FacesRatingBar/FacesRatingBar.js:213]: Just for this file to be clean, can you create a folder with these icons and import them?

File: src/FacesRatingBar/FacesRatingBar.js [hunk 8]
```diff
@@ -0,0 +1,327 @@
...
+  };
+
+  render() {
+    const { dataHook, readOnly, value, descriptionValues } = this.props;
+    const { faceHoveredIndex } = this.state;
+
+    const showDescriptionValues = this._shouldShowDescriptionValues();
+
+    return (
+      <Box {...styles('root', {}, this.props)} data-hook={dataHook}>
```
[the remark this answers is not in the document]
Comment (EgoziE) [reply src/FacesRatingBar/FacesRatingBar.js:73]: This makes the smoke test fail

File: src/FacesRatingBar/FacesRatingBar.js (hunk 7 above)
Comment (sivanharel) [reply src/FacesRatingBar/FacesRatingBar.js:213]: Is it ok to create one js file for all of them or should I create a file for each icon?

**Pull request merged**